skills/ssm-executor/SKILL.md

```diff
@@ -0,0 +1,179 @@
+---
+name: ssm-executor
+description: Execute commands on any Bottlerocket EC2 instance via SSM Session Manager
+---
+
+# Skill: SSM Executor
+
+## Purpose
+
+Execute commands on Bottlerocket EC2 instances using AWS Systems Manager (SSM), without requiring Kubernetes or direct SSH access. Works with any Bottlerocket variant (ECS, K8s, standalone).
+
+## When to Use
+
+- Debugging Bottlerocket instances without K8s access
+- Running commands on ECS or standalone Bottlerocket nodes
+- Inspecting host configuration via apiclient
+- Accessing the host rootfs via sheltie
+
+## Prerequisites
+
+- AWS CLI configured with appropriate permissions
+- SSM agent running on the instance (enabled by default on Bottlerocket)
+- Instance must have SSM connectivity (via VPC endpoint or internet access)
+- IAM instance profile with SSM permissions
+
+## Procedure
+
+### 1. Verify SSM Connectivity
+
+```bash
+aws ssm describe-instance-information \
+  --filters Key=InstanceIds,Values=INSTANCE_ID \
+  --query 'InstanceInformationList[*].[InstanceId,PingStatus,PlatformName]' \
+  --output table --region REGION
+```
+
+Expected: `Online` status and `Bottlerocket` platform.
+
+### 2. Execute Commands
+
+**Simple command (control container context):**
+```bash
+CMD_ID=$(aws ssm send-command \
+  --instance-ids INSTANCE_ID \
+  --document-name "AWS-RunShellScript" \
+  --parameters '{"commands":["uname -a"]}' \
+  --region REGION \
+  --query 'Command.CommandId' \
+  --output text) && \
+sleep 3 && \
+aws ssm get-command-invocation \
+  --command-id $CMD_ID \
+  --instance-id INSTANCE_ID \
+  --region REGION \
+  --query 'StandardOutputContent' \
+  --output text
+```
+
+**Access host rootfs via sheltie (full host access):**
+```bash
+CMD_ID=$(aws ssm send-command \
+  --instance-ids INSTANCE_ID \
+  --document-name "AWS-RunShellScript" \
+  --parameters '{"commands":["apiclient exec admin sheltie -- containerd --version"]}' \
+  --region REGION \
+  --query 'Command.CommandId' \
+  --output text) && \
+sleep 4 && \
+aws ssm get-command-invocation \
+  --command-id $CMD_ID \
+  --instance-id INSTANCE_ID \
+  --region REGION \
+  --query 'StandardOutputContent' \
+  --output text
+```
+
+### 3. Understanding the Execution Context
+
+SSM commands run through a chain of contexts:
+
+| Context | Access | How to Reach |
+|---------|--------|-------------|
+| Control container | SSM agent, apiclient | Direct SSM command |
+| Admin container | AL2023 userspace, /.bottlerocket/rootfs | `apiclient exec admin` |
+| Host namespace | Full rootfs, all binaries | `apiclient exec admin sheltie --` |
```

Comment:
I think this implies we need something under $FOREST/docs to talk about these contexts. It's not clear to me if all agents working in the forest should just know it, or if we should make it dynamically discoverable. BUT this is very useful information that should be hoisted and shared.